### DISCUSSION TITLE
Anzahl HM / HMIP Geräte ermitteln

### DISCUSSION THREAD

**Discussion by manny4566** [OpenCCU/OpenCCU#2657] [🇩🇪 Fragen&Antworten]
Hallo,

gibt es eine einfache Möglichkeit die aktuelle Anzahl an angelernten HM oder HMIP Geräten zu ermitteln?
Falls noch nicht vorhanden würde ich mir z.b. auf der Status Seite (Button Hilfe) wünschen, dass dort vielleicht ein Counter mit aufgeführt wird um zu zeigen wie viele HM/HMIP Geräte aktuell angelernt sind (ggf. Feature Request?)

P.S. Falls das Feature erwünscht ist, könnte man das auch noch so weit treiben, dass die Geräteanzahl pro Typ auch noch zusätzlich aufsummiert wird um einen Überblick zu verschaffen.
Danke

**Comment by ghost** [comment]
abgespeckte Ausführung:
`string sChan;object oChan;object oDev;integer AnzHM = 0;integer AnzIP = 0;
foreach(sChan, channels.EnumIDs()){oChan = dom.GetObject(sChan);
if (oChan.Interface() <> ID_ERROR){oDev = dom.GetObject(oChan.Device());
if (true){if ((oDev.Label() <> "HmIP-RCV-50") && (oDev.Label() <> "HM-RCV-50")){
if (oChan.ChnNumber() == 0){
if (oDev.Label().StartsWith("HmIP") || oDev.Label().StartsWith("HMIP")){ AnzIP = AnzIP + 1;}else{AnzHM = AnzHM + 1;}}}}}}
WriteLine("Anzahl HM-Geräte   : " # AnzHM);
WriteLine("Anzahl HMIP-Geräte : " # AnzIP);`

der Autor ist "Tyfys" [link](https://homematic-forum.de/forum/viewtopic.php?f=59&t=70752&hilit=Auslesen+Seriennummer+und+Barcode&start=20)
